tests/test_imputation/test_tmi_imputation.py

```diff
@@ -0,0 +1,179 @@
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from src.imputation.tmi_imputation import (
+    create_mean_dict,
+    apply_tmi,
+)
+
+#testing
+def test_create_mean_dict():
+
+    cols = [
+        "statusencoded",
+        "imp_class",
+        "Value_trim",
+        "Value1",
+        "Value2",
+        "reference",
+        "employees",
+        "formtype",
+    ]
+
+    # Create a sample DataFrame for testing
+    data = [
+        ["210", "A", "dont trim", None, 100, "A", 100, "0001"],
```

Comment:
I ran the test with the other checks, and the test fails when the None is replaced by an empty string ('nan'  was fine and passed) - I'm getting a TypeError: '>' not supported between instances of 'str' and 'int'